File: extractall.py
import os
import matlab


def extractall():
    # run mfccextract on all wav files in recordings
    # returns 0 if 1 or more wav files found

    eng = matlab.engine.start_matlab()

    count = 0

    for root, dirs, files in os.walk(os.getcwd() + '/data/recordings'):
        for file in files:
            if file.endswith('.wav'):
                print('extracting features from ' + file + '\n...')
                print(eng.mfccextract(file) + 'created')
                count += 1

    print('finished\nfiles extracted: ' + count)

    return 0 if count else 1

File: extractall.m
function [extracted, size] = extractall(fDir)
% This function runs mfccextract on all wav files
% in directories within fDir
% returns true if any wav file found else false
sDir = pwd;
cd(fDir);
D = dir;
extracted = false;

for k = 3:length(D)
    currD = D(k).name;
    if ~startsWith(currD, '.')
        cd(fDir);
        cd(currD);
        disp('Checking subdirectory:');
        disp(currD);
        fList = (dir('*.wav'));
        for x = 1:length(fList)
            disp('Extracting features from:');
            f = fList(x).name;
            disp(f);
            size = mfccextract(f);
            extracted = true;
        end
        cd('..');    
    end
cd(sDir);
end
